Lib/subprocess.py

```diff
@@ -1420,9 +1428,23 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                     # the ones in the handle_list
                     close_fds = False
 
-            if shell:
+            if force_hide or shell:
+                # We pass SW_HIDE to the process so that it will not display any
+                # window even if it normally would.
                 startupinfo.dwFlags |= _winapi.STARTF_USESHOWWINDOW
                 startupinfo.wShowWindow = _winapi.SW_HIDE
+
+            if force_hide and not (creationflags & _winapi.DETACHED_PROCESS):
+                # If the child process is flagged as SUBSYSTEM_CONSOLE, then
+                # it either inherits the console of its parent, if the parent
+                # has one, or allocates a new console. An inherited console may
+                # be visible even if SW_HIDE is set. By setting the creation
+                # flag CREATE_NEW_CONSOLE, the child is forced to allocate a
+                # new console that will have a hidden window.
+                # Note: CREATE_NEW_CONSOLE cannot be used with DETACHED_PROCESS.
```

Comment:
I know I previously suggested this, but on second reading it provides too much information in some ways and not enough in other ways, and it's just too long. How about this instead?
```suggestion
                # The startupinfo show-window command applies only to the first
                # window that an application creates, so force console apps to
                # create a new console via CREATE_NEW_CONSOLE. This flag is
                # ignored by GUI apps and cannot be used with DETACHED_PROCESS.
```